Automatically detect and exit copy mode when sending commands #70

vinodkri opened the pull request:
Hi Chris,

As with the previous issue; I needed this feature support as well. For now, I have only handled the "copy-mode" exit.
I assume the following issue "Stop current running process in runner #79" will be handled as part of the VtrSendKeysRaw feature.

christoomey commented:
@vinodkri thanks so much for all the changes. I reworked the code a tiny bit to extract some of the logic from the outermost `AttachToPane` function (and renamed it to match), but the implementation is the same as you had it. I've also cherry picked the copy mode command and merged that in, all with you as the author.

Thanks again for taking these on! I've wanted all of this functionality for a while and am really glad to have it now.

vinodkri commented:
Hi Chris,

The changes looks cool! I wish I would have written it that way. Thanks for
pulling the copy mode exit PR.

In the function s:AttachToSpecifiedPane below two lines look stale to me!.
let s:vim_pane = s:ActivePaneIndex()
let s:vtr_orientation = s:CurrentMajorOrientation()

Cheers,
Vinod

On Mon 16 Apr, 2018, 10:58 PM Chris Toomey, <notifications@github.com>
wrote:

> @vinodkri <https://github.com/vinodkri> thanks so much for all the
> changes. I reworked the code a tiny bit to extract some of the logic from
> the outermost AttachToPane function (and renamed it to match), but the
> implementation is the same as you had it. I've also cherry picked the copy
> mode command and merged that in, all with you as the author.
>
> Thanks again for taking these on! I've wanted all of this functionality
> for a while and am really glad to have it now.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/christoomey/vim-tmux-runner/pull/85#issuecomment-381684493>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIEqzdOmZr5YhW-UkpEnVU3xh1jmgjszks5tpNS1gaJpZM4TWLuN>
> .
>

vinodkri commented:
Hey Chris,

On second note. I understand the usage of those two lines now.
Ignore the comment :)

Cheers,
Vinod

On Tue, Apr 17, 2018 at 7:28 AM, Vinod Krishna <
vinod.krishna.bangalore@gmail.com> wrote:

> Hi Chris,
>
> The changes looks cool! I wish I would have written it that way. Thanks
> for pulling the copy mode exit PR.
>
> In the function s:AttachToSpecifiedPane below two lines look stale to me!.
> let s:vim_pane = s:ActivePaneIndex()
> let s:vtr_orientation = s:CurrentMajorOrientation()
>
> Cheers,
> Vinod
>
>
> On Mon 16 Apr, 2018, 10:58 PM Chris Toomey, <notifications@github.com>
> wrote:
>
>> @vinodkri <https://github.com/vinodkri> thanks so much for all the
>> changes. I reworked the code a tiny bit to extract some of the logic from
>> the outermost AttachToPane function (and renamed it to match), but the
>> implementation is the same as you had it. I've also cherry picked the copy
>> mode command and merged that in, all with you as the author.
>>
>> Thanks again for taking these on! I've wanted all of this functionality
>> for a while and am really glad to have it now.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/christoomey/vim-tmux-runner/pull/85#issuecomment-381684493>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AIEqzdOmZr5YhW-UkpEnVU3xh1jmgjszks5tpNS1gaJpZM4TWLuN>
>> .
>>
>


-- 
*Warm Regards,*
*Vinod Krishna*
